lib/ecto/repo/model.ex

```diff
@@ -150,11 +165,13 @@ defmodule Ecto.Repo.Model do
     Planner.fields(kind, model, Map.take(changeset.changes, fields))
   end
 
-  defp pk_filter(model, struct) do
+  defp filters!(model, struct, %Changeset{filters: filters}) do
+    cs_filters = Enum.to_list(filters)
```

Comment:
Don't covert to a list. Simply add the primary key using `Map.put` and let `Planner.fields` do the conversion once it iterates.